src/main/java/de/qabel/desktop/ui/contact/menu/ContactMenuController.java

```diff
@@ -0,0 +1,174 @@
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Button;
+import javafx.scene.control.Tooltip;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.VBox;
+import javafx.stage.FileChooser;
+import org.controlsfx.control.PopOver;
+import org.json.JSONException;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.*;
+
+public class ContactMenuController extends AbstractController implements Initializable {
+
+    ResourceBundle resourceBundle;
+
+    @Inject
+    private ContactRepository contactRepository;
+
+    @Inject
+    private ClientConfig clientConfiguration;
+
+    @Inject
+    Pane layoutWindow;
+
+    @Inject
+    private TransactionManager transactionManager;
+
+    @FXML
+    AnchorPane menuContact;
+
+    @FXML
+    VBox vboxMenu;
+
+    @FXML
+    Button importButton;
+
+    @FXML
+    Button exportToFileButton;
+
+    @FXML
+    Button importFromFile;
+    @FXML
+    Button importFromQR;
+    @FXML
+    Button searchButton;
+    @FXML
+    Button enterContact;
+    @FXML
+    Button exportContactsToFile;
+    @FXML
+    Button exportContactsToQR;
+
+    Identity identity;
+
+    ImageView imageView;
+
+    public PopOver popOver;
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        resourceBundle = resources;
+        identity = clientConfiguration.getSelectedIdentity();
+    }
+
+    public void showMenu(double coordX, double coordY) {
+        initializePopOver();
+        createButtonGraphics();
+        popOver.show(menuContact, coordX, coordY);
+    }
+
+    private void createButtonGraphics() {
+        Tooltip.install(importButton, new Tooltip(resourceBundle.getString("contactImport")));
+        Tooltip.install(exportToFileButton, new Tooltip(resourceBundle.getString("contactExport")));
+        importFromFile.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/import_black.png"))));
+        importFromQR.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/qrcode.png"))));
+        searchButton.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/cloud_search.png"))));
+        enterContact.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/pencil.png"))));
+        exportContactsToFile.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/export.png"))));
+        exportContactsToQR.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/qrcode.png"))));
+    }
+
+    private ImageView getImage(Image image) {
+        imageView = new ImageView();
+        imageView.setImage(image);
+        return imageView;
+    }
+
+    private void initializePopOver() {
+        popOver = new PopOver();
```

Comment:
You are actually doing this the wrong way.
The ContactMenuController shouldn't know that it's displayed inside a PopOver .. it actually can't know (in a test you might for example display it in a normal pane instead of a popover).
The problem starts in the controller where you do
`contactMenuView.getView(layoutWindow.getChildren()::add)` ... where you add a view that you later have to remove because you don't want it there ....
If you would do a 
```JAVA
contactMenuView.getView( view -> {
    popover = new PopOver(view);
    popover.show(contactRootItem);
})
```
you would be fine and the contactMenuController wouldn't have to know anything about popovers.